Content.Server/Medical/CryoPodSystem.cs

```diff
@@ -0,0 +1,389 @@
+        {
+            _popupSystem.PopupEntity(Loc.GetString("cryo-pod-locked"), uid, Filter.Entities(userId));
+            return;
+        }
+
+        EjectBody(uid, cryoPodComponent);
+    }
+
+    public void EjectBody(EntityUid uid, CryoPodComponent? cryoPodComponent)
+    {
+        if (!Resolve(uid, ref cryoPodComponent))
+            return;
+
+        if (cryoPodComponent.BodyContainer.ContainedEntity is not {Valid: true} contained)
+            return;
+
+        cryoPodComponent.BodyContainer.Remove(contained);
+        RemComp<InsideCryoPodComponent>(contained);
+        _climbSystem.ForciblySetClimbing(contained, uid);
+
+        // Restore the correct position of the patient. Checking the components manually feels hacky, but I did not find a better way for now.
+        if (HasComp<KnockedDownComponent>(contained) || _mobStateSystem.IsIncapacitated(contained))
+        {
+            _standingStateSystem.Down(contained);
+        }
+        else
+        {
+            _standingStateSystem.Stand(contained);
+        }
+
+        UpdateAppearance(uid, cryoPodComponent);
+    }
+
+    #region Interaction
+
+    private void HandleDragDropOn(EntityUid uid, CryoPodComponent cryoPodComponent, DragDropEvent args)
+    {
+        if (cryoPodComponent.BodyContainer.ContainedEntity != null)
+        {
+            return;
+        }
+
+        var doAfterArgs = new DoAfterEventArgs(args.User, cryoPodComponent.EntryDelay, default, uid, args.Dragged)
+        {
+            BreakOnDamage = true,
+            BreakOnStun = true,
+            BreakOnTargetMove = true,
+            BreakOnUserMove = true,
+            NeedHand = false,
+            TargetFinishedEvent = new DoInsertCryoPodEvent(args.Dragged),
+        };
+        _doAfterSystem.DoAfter(doAfterArgs);
+        args.Handled = true;
+    }
+
+    private void DoInsertCryoPod(EntityUid uid, CryoPodComponent cryoPodComponent, DoInsertCryoPodEvent args)
+    {
+        InsertBody(uid, args.ToInsert, cryoPodComponent);
+    }
+
+    private void HandleInteractHand(EntityUid uid, CryoPodComponent cryoPodComponent, InteractHandEvent args)
+    {
+        if (!HasComp<ActiveCryoPodComponent>(uid))
+        {
+            return;
+        }
+        if (!TryComp(args.User, out ActorComponent? actor))
+            return;
+
+        if (cryoPodComponent.BodyContainer.ContainedEntity != null)
+        {
+            if (args.User == cryoPodComponent.BodyContainer.ContainedEntity)
+            {
+                return;
+            }
+            _userInterfaceSystem.TryOpen(uid, SharedHealthAnalyzerComponent.HealthAnalyzerUiKey.Key, actor.PlayerSession);
+            _userInterfaceSystem.TrySendUiMessage(
+                uid,
+                SharedHealthAnalyzerComponent.HealthAnalyzerUiKey.Key,
+                new SharedHealthAnalyzerComponent.HealthAnalyzerScannedUserMessage(cryoPodComponent.BodyContainer.ContainedEntity));
+            args.Handled = true;
+        }
+    }
```

Comment:
use the `ActivatableUI` component in yaml for this instead to avoid duplication. you can subscribe to some 'after activatable UI open' (i think) event to do the UI message sending, and theres a before UI open attempt event to cancel if its not active etc